spec/nxt_state_machine_spec.rb

```diff
@@ -126,6 +126,45 @@
     end
   end
 
+  describe '#state_machine' do
+    subject do
+      ArticleWorkflow.new(article, test: 'options').state_machine
+    end
+
+    describe '#states' do
+      it 'returns all states' do
+        expect(subject.states.keys).to match_array(%w[
+          draft
+          written
+          submitted
+          approved
+          published
+          rejected
+          deleted
+        ])
+      end
+    end
+
+    describe '#event_methods' do
+      it 'returns all event names and bang versions' do
+        expect(subject.event_methods).to match_array([
+          :write,
+          :write!,
+          :submit,
+          :submit!,
+          :approve,
+          :approve!,
+          :publish,
+          :publish!,
+          :reject,
```

Comment:
In fact we'll have a different order here:

```ruby
[
  :write,
  :submit,
  :approve,
  :write!,
  :submit!,
  :approve!,
]
```
match_array() will mitigate this so I'm mostly wondering if it matters for the end user or not. 🤔

Reply:
Yeah, it hardly matters. This was just me being lazy, but I'll change the spec to reflect reality.